lib/signs_ui/signs/state.ex

```diff
@@ -20,16 +20,25 @@ defmodule SignsUI.Signs.State do
   end
 
   def init(_) do
-    signs_url = Application.get_env(:signs_ui, :signs_url)
-    signs = Signs.Request.get_signs(signs_url)
-    {:ok, signs}
+    Signs.Request.get_signs()
```

Comment:
`get_signs/0` here returns `{:ok, any()} | {:error, any()}` , which is right since it can either succeed or fail.

However, the GenServer init callback has a [different spec](https://hexdocs.pm/elixir/GenServer.html#c:init/1). If it returns `{:ok, state}` then the GenServer starts up with that initial state. But if you want to signal that the GenServer failed to start, you need to return `{:stop, reason}`.

I think it's correct for `get_signs` to return `{:error, ...}`, you should just handle it here and return either `{:ok, ...}` or `{:stop, ...}`.